x/perpetual/keeper/get_net_open_interest.go

```diff
@@ -6,16 +6,68 @@ import (
 	"github.com/elys-network/elys/x/perpetual/types"
 )
 
-func (k Keeper) GetNetOpenInterest(pool types.Pool) math.Int {
+func (k Keeper) GetNetOpenInterest(ctx sdk.Context, pool types.Pool) math.Int {
 	assetLiabilitiesLong := sdk.ZeroInt()
 	assetLiabilitiesShort := sdk.ZeroInt()
 
+	uusdc, found := k.assetProfileKeeper.GetEntry(ctx, "uusdc")
+
+	if !found {
+		return sdk.ZeroInt()
+	}
+
 	for _, asset := range pool.PoolAssetsLong {
-		assetLiabilitiesLong = assetLiabilitiesLong.Add(asset.Liabilities)
+
+		if asset.Liabilities.IsZero() {
+			continue
+		}
+
+		if uusdc.Denom == asset.AssetDenom {
+			assetLiabilitiesLong = assetLiabilitiesLong.Add(asset.Liabilities)
+		} else {
+
+			coin := sdk.NewCoin(asset.AssetDenom, asset.Liabilities)
+			ammPool, err := k.GetAmmPool(ctx, pool.AmmPoolId, asset.AssetDenom)
+
+			if err != nil {
+				return sdk.ZeroInt()
+			}
+
+			estimatedSwap, err := k.OpenLongChecker.EstimateSwapGivenOut(ctx, coin, uusdc.Denom, ammPool)
+			if err != nil {
+				return sdk.ZeroInt()
+			}
+
+			assetLiabilitiesLong = assetLiabilitiesLong.Add(estimatedSwap)
+		}
+
```

Comment:
@fenriz07 you seem to be repeating the same code at many places, could you create a separate function that is called instead, can you also apply those changes to here as @amityadav0 mentioned, thanks

Reply:
@fenriz07 @amityadav0 updated PR with the changes mentioned above include `settle_funding_fee_distribution`